### PR TITLE
[UI/UX] New icon pack

### LINKED ISSUE #13138: Add a new set of icons
### Description
Introducing a new set of icons will refresh the platform's visual elements and potentially enhance user comprehension and engagement.

### Impact
The addition of a new set of icons could positively impact user experience by making the interface more visually appealing and intuitive, potentially reducing the learning curve and increasing user engagement.

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

### PR DESCRIPTION
**Pull request by andersonjeccel**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #13138  

#### Description:

Introducing a new set of icons will refresh the platform's visual elements and potentially enhance user comprehension and engagement.

It will be used in next PRs.

The addition of a new set of icons could positively impact user experience by making the interface more visually appealing and intuitive, potentially reducing the learning curve and increasing user engagement. Furthermore, the number of icons available in the current Font Awesome icon pack is less than 700, causing the use of icons for buttons without effectively representing actions in the interface. 

OBS.! If we were to update Font Awesome, we would have to subscribe to a monthly paid subscription to access the full package as it is not entirely free. Remix is ​​100% free and continually growing.

Initially I considered all of these open source icon packs:
Material Icons, Iconpark Icons, Remix Icons, Phosphor Icons, Carbon Icons, Bootstrap Icons, Boxicons, Tabler Icons, Lineawesome Icons, Ionic Icons, Emblem Icons, Teeny Icons, Iconoir Icons, Jam Icons, Cssgg Icons, Majestic Icons, Lucide Icons, Kalai Icons, Basil Icons, Pixelart Icons, Doodle Icons, Heroicons, Systemui Icons, Coolicons, Codicons, Eva Icons, Akar Icons, Radix Icons, Ikonate Icons, Feather Icons, Basicons, Iconic Icons, Mono Icons, Pep Icons, Bytesize Icons

Then I started to filter by quantity (only the ones that have at least 1000 icons) and license compatibility. This reduced the list to:
Material Icons, Iconpark Icons, Remix Icons, Phosphor Icons, Carbon Icons, Bootstrap Icons, Boxicons, Tabler Icons, Lineawesome Icons, Ionic Icons, Emblem Icons, Teeny Icons, Iconoir Icons

Preference was given to packs that offer filled icons, not just line icons. The list was further reduced to:
Material Icons, Remix Icons, Phosphor Icons, Carbon Icons, Bootstrap Icons, Boxicons, Ionic Icons, Emblem Icons, Teeny Icons

Among similar icon packs, those with a larger quantity were preferred. This led to the elimination of Emblem Icons and Teeny Icons, which are very similar to Material Icons and Remix Icons, respectively, but have fewer icons. Phosphor was also eliminated as it is similar to the current icon pack.

Remaining: Material, Remix, Carbon, Bootstrap, Ionic

Then one by one filtering:

Carbon Icons: it has 1700 more line icons than solid (~150), so we could run out of icons in scenarios where the icon need to be tiny (solid icons are better to see when small)

Ionic Icons: Eliminated due to inflated numbers - half of the icons simply had a rounded line instead of sharp, reducing the total number of usable icons to less than 1000.

Bootstrap Icons: Eliminated due to having twice as many line (~1200) icons as solid (~500) ones, which could again lead to visibility issues given that the Mautic interface is built to use small solid icons.

Material Icons: Although this pack has more than 1000 line and solid icons each, implementing them on Mautic would be tricky as it depends on repositories not maintained by Google, which don’t offer ready LESS or CSS files as Mautic uses.

Remix Icons: This pack has a small difference between the number of line and solid icons, with both types having more than 1000 icons each. It also offers ready font, LESS, and CSS files, making it easy to update (just throw the new file and you're good to go)

Preview

#### Steps to test this PR:

1. Only review the code, since changes will be made in separate PRs

### CODE REVIEW

**Comment by LordRembo** [comment]
Just thinking practically, given what has been discussed before, maybe it makes more sense to:
- make PR's for icon changes against a new `5.x-ui-redesign` branch, instead of against `5.x` (I believe this was brought up before, to make incremental change + overall big changes possible)
- close this PR (which only adds the new icon font), and focus on merging individual PR's that actually replace 1 or more icons [like this one for the right sidebar icon](https://github.com/mautic/mautic/pull/13365). These already include the necessary icon fonts.

@RCheesley @mollux if you can agree with item 1 of that flow, can one of you make that UI branch on the Mautic repo so @andersonjeccel can continue his work?

**Comment by andersonjeccel** [comment]
> * make PR's for icon changes against a new `5.x-ui-redesign` branch, instead of against `5.x` (I believe this was brought up before, to make incremental change + overall big changes possible)
> * close this PR (which only adds the new icon font), and focus on merging individual PR's that actually replace 1 or more icons [like this one for the right sidebar icon](https://github.com/mautic/mautic/pull/13365). These already include the necessary icon fonts.


Oh, the inclusion of the icon pack in the sidebar PR was an accident that I later undid. I believe it would be necessary to merge both into the main branch...

If you prefer that I actually proceed to do this merge into the redesign branch, I will have to change both (I believe it may not be beneficial to let these UX changes be made together with a future UI as discussed with Mike).

**Comment by LordRembo** [comment]
@andersonjeccel Ah, okay, I see. My logic was that, it was first and foremost a UI change and a UX change secondary. Given it changes actual design elements and the changes are part of a bigger overall change (replacing ALL icons). It would probably get merged into a UI branch faster than it'll get into the 5.x branch. Since the UI branch is targeted for a 5.1 release and the 5.x branch is also used for the minor releases.
But I don't want to further confuse or complicate the flow if it's already been discussed before. Let's let the PR be active as it is for now then, and I'll bring it up on our next meeting with Mike to make sure we're all on the same page.

For the second item: by not separating the inclusion of the icon files/less but instead just having those already in each individual 'icon change' PR, you wouldn't have to wait for the first PR to get merged. I don't really like the idea of having a PR that depends on another PR if it can be avoided.

**Comment by LordRembo** [comment]
But again, that's something we can discuss next week.

**Comment by andersonjeccel** [comment]
> I don't really like the idea of having a PR that depends on another PR if it can be avoided.

Hm, I'm actually following the developers' request to separate each change into its own pull request to make code review easier. There is also #13375 that depends on it...

I assume they were waiting for practical scenarios where the icons were being applied before definitively merging the pull request. Idk, perhaps we can just wait some days or their instructions, but I need this merged to start replacing the old icons

**Review by LordRembo — APPROVED** [review]
@andersonjeccel Alrighty, if it's an agreed upon system, I'm gonna stay out of it. Code looks fine, makes sense to load it here. It's got a positive review from me in any case.

**Review by escopecz — COMMENTED** [review]
Would it be possible to [install the library via NPM](https://github.com/Remix-Design/RemixIcon?tab=readme-ov-file#installation)?

That would then make the upgrades easier and dependabot would report and fix the security issues if any occur in this version.

**Comment by andersonjeccel** [comment]
> Would it be possible to [install the library via NPM](https://github.com/Remix-Design/RemixIcon?tab=readme-ov-file#installation)?

Hm, it's working when I try to use an icon. Are the new commits ok?

**Review by escopecz — APPROVED** [review]
The code changes look great to me 👍

File: app/bundles/CoreBundle/Assets/css/libraries/libraries.less
Comment (adiux) [inline app/bundles/CoreBundle/Assets/css/libraries/libraries.less:7]: Untested, but I think this could be changed to just `remixicon/fonts/remixicon`?

File: app/bundles/CoreBundle/Assets/css/libraries/libraries.css
Comment (adiux) [inline app/bundles/CoreBundle/Assets/css/libraries/libraries.css:2]: Untested, but I think this could be changed to just `remixicon/fonts/remixicon`? this way it is more robust against path changes and in the same style as other imports.

Comment (tomfriedhof) [reply app/bundles/CoreBundle/Assets/css/libraries/libraries.css:2]: I just tested this, it works and it's a better solution. The current solution does not work well in prod mode after running `mautic:assets:generate`. The styles in `remixicon.css` do not get expanded into the `libraries.css` file when the generate command is run.

Comment (tomfriedhof) [reply app/bundles/CoreBundle/Assets/css/libraries/libraries.css:2]: Actually, I take that back. When importing with `remixicon/fonts/remixicon` instead of with `../../../../../../node_modules/remixicon/fonts/remixicon.css` it wrote the css into the `libraries.css` but for some reason the browser didn't apply the styles for the "Settings" icon.

I ended up reverting the `@import` back and moved the `node_modules` directory into my `docroot` so that it could resolve the import.

**Pull request merged**
